tests/functional/disaster-recovery/sc_arbiter/test_zone_shutdown_and_crash.py

```diff
@@ -515,77 +401,93 @@ def test_zone_crashes(
-
-        assert sc_obj.check_for_data_loss(
-            constants.LOGWRITER_CEPHFS_LABEL
-        ), "[CephFS] Data is lost"
-        log.info("[CephFS] No data loss is seen")
-        assert sc_obj.check_for_data_loss(
-            constants.LOGWRITER_RBD_LABEL
-        ), "[RBD] Data is lost"
-        log.info("[RBD] No data loss is seen")
+        verify_data_loss(sc_obj)
 
         # check for data corruption
         sc_obj.cephfs_logreader_job.delete()
+        log.info(sc_obj.cephfs_logreader_pods)
         for pod in sc_obj.cephfs_logreader_pods:
             pod.wait_for_pod_delete(timeout=120)
-        log.info("All old logreader pods are deleted")
-        pvc = get_pvc_objs(
-            pvc_names=[
-                sc_obj.cephfs_logwriter_dep.get()["spec"]["template"]["spec"][
-                    "volumes"
-                ][0]["persistentVolumeClaim"]["claimName"]
-            ],
-            namespace=constants.STRETCH_CLUSTER_NAMESPACE,
-        )[0]
-        logreader_workload_factory(
-            pvc=pvc, logreader_path=constants.LOGWRITER_CEPHFS_READER, duration=5
+        log.info("All old CephFS logreader pods are deleted")
+        verify_data_corruption(sc_obj, logreader_workload_factory)
+
+
+class TestGracefulRestart:
+
+    def test_gracefull_restart(
+        self,
+        init_sanity,
+        node_drain_teardown,
+        reset_conn_score,
+        setup_logwriter_cephfs_workload_factory,
+        setup_logwriter_rbd_workload_factory,
+        logreader_workload_factory,
+        nodes,
+        cnv_workload,
+        setup_cnv,
+        # node_types,
+    ):
+        """
+        Test node grace full shutdown while the workloads are Running
+
+        """
+
+        sc_obj = StretchCluster()
+
+        # Run the logwriter cephFs workloads
+        log.info("Running logwriter cephFS and RBD workloads")
+        (
+            sc_obj.cephfs_logwriter_dep,
+            sc_obj.cephfs_logreader_job,
+        ) = setup_logwriter_cephfs_workload_factory(read_duration=0)
+        sc_obj.rbd_logwriter_sts = setup_logwriter_rbd_workload_factory(
+            zone_aware=False
         )
 
-        sc_obj.get_logwriter_reader_pods(constants.LOGREADER_CEPHFS_LABEL)
+        # setup vm and write some data to the VM instance
+        vm_obj = cnv_workload(volume_interface=constants.VM_VOLUME_PVC)
+        vm_obj.run_ssh_cmd(
+            command="dd if=/dev/zero of=/file_1.txt bs=1024 count=102400"
+        )
+        md5sum_before = cal_md5sum_vm(vm_obj, file_path="/file_1.txt")
+
+        # make sure all the worload pods are running
+        check_for_logwriter_workload_pods(sc_obj, nodes=nodes)
+        log.info("All logwriter workload pods are running successfully")
+
+        gracefully_reboot_nodes()
+        log.info("Gracefully restarted all the cluster nodes")
+
+        time.sleep(60)
+        log.info("a moment to breathe!")
+
+        # wait for all storage pods to be running or completed
```

Comment:
we need to check for the nodes to be in ready state before checking the pods status

Reply:
@PrasadDesala thats being taken care by the graceful_restart_nodes () function